dashboard/test/ui/test_status.js

```diff
@@ -102,7 +109,10 @@ Test.prototype.fetchStatus = function () {
         this.status = this.success ? STATUS_SUCCEEDED : STATUS_FAILED;
       }
     })
-    .then(ensure, ensure);
+    .then(ensure, error => {
+      ensure();
+      throw error;
+    });
```

Comment:
Runs our `ensure` block (in this case set `isUpdating=false` and rerender) in both the failure and success cases, but rethrow our error so that it bubbles up the promise chain to the `Promise.all`, below.